### ISSUE TITLE
Can't seem to connect to SMTP Server

### ISSUE THREAD

**Issue by KaloyJS** [PHPMailer/PHPMailer#2515]
Hi I tried to upgrade from 5.2, my credentials are ok and everything is working in 5.2

my settings:
`$mail->SMTPDebug = SMTP::DEBUG_SERVER;                      //Enable verbose debug output
        $mail->isSMTP();                                            //Send using SMTP
        $mail->Host       = 'server';                     //Set the SMTP server to send through
        $mail->SMTPAuth   = true;                                   //Enable SMTP authentication
        $mail->Username   = "username";                     //SMTP username
        $mail->Password   = "password";                               //SMTP password        
        $mail->SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS;
        $mail->Port       = 587;`

and error message is this:
2021-10-01 19:20:46 SERVER -> CLIENT: 220 zimbra.sbe-ltd.co.uk ESMTP Postfix
2021-10-01 19:20:46 CLIENT -> SERVER: EHLO portal-ca.sbe-ltd.ca
2021-10-01 19:20:46 SERVER -> CLIENT: 250-zimbra.sbe-ltd.co.uk250-PIPELINING250-SIZE 52428800250-VRFY250-ETRN250-STARTTLS250-AUTH PLAIN LOGIN250-AUTH=PLAIN LOGIN250-ENHANCEDSTATUSCODES250-8BITMIME250-DSN250 CHUNKING
2021-10-01 19:20:46 CLIENT -> SERVER: STARTTLS
2021-10-01 19:20:46 SERVER -> CLIENT: 220 2.0.0 Ready to start TLS
SMTP Error: Could not connect to SMTP host.
2021-10-01 19:20:46 CLIENT -> SERVER: QUIT
2021-10-01 19:20:46 SERVER -> CLIENT:
2021-10-01 19:20:46 SMTP ERROR: QUIT command failed:
SMTP Error: Could not connect to SMTP host.
Message could not be sent. Mailer Error: SMTP Error: Could not connect to SMTP host.

Thanks in advance...

**Comment by Synchro** [comment]
Is it also running on the same instance & version of PHP? This is discussed extensively in the troubleshooting guide – what did the openssl tests tell you?

**Comment by chroust** [comment]
@KaloyJS  I have the same problem.
I found out there is a problem with Let`s Encrypt certificates

**Comment by KaloyJS** [comment]
@Synchro yes its running on the same php 5.6 instance. I haven't run the openssl test I will get back to you with that.
@Chroust can you share to me how to check if that is the same as my situation.

**Comment by KaloyJS** [comment]
@chroust thanks, it did sent without SSL/TLS. How did you solve the certification issue thanks in advance ^^

**Comment by KaloyJS** [comment]
@Synchro  thank you for the reply I will try to work on this solution.

**Comment by chroust** [comment]
@KaloyJS It has not been resolved yet. Hosting support works to fix the problem.

**Comment by supick** [comment]
`yum install ca-certificates` fixed this issue for centos 7

**Comment by TLWebdesign** [comment]
> `yum install ca-certificates` fixed this issue for centos 7

This also worked for me. Thanks!

**Comment by david-littlefield** [comment]
Hi @Synchro, you are the most responsive and active contributor that I've seen. I combed through the issues section, and read through the troubleshooting guide. I wanted to add something that might have been mentioned in another post - I can't remember, and I couldn't find it again. I wasn't able to connect to the smtp host because my cloud service provider blocks the port 25, 465, and 587 by default to prevent potential spam. This can be removed by asking customer service to do so. They will issue a short checklist of simple requirements to perform to unlock those ports for all future virtual machines.

If it's ok, I have a question about handling emails in production. I've seen your suggestion about [maximizing performance](https://github.com/PHPMailer/PHPMailer/wiki/Sending-to-lists#maximising-performance). Unfortunately, I found your suggestion about avoiding [rabbitmq](https://github.com/PHPMailer/PHPMailer/issues/1153#issuecomment-351931042) after consuming several courses, books, and articles on it, haha. With the goal of quick page load times, and making use of the sunk learning cost, would it make sense to pass the phpmailer object to a server that routes it to rabbitmq on the server's localhost?

**Comment by Synchro** [comment]
Thank you for your kind words! There's nothing wrong with using queues in general, but using queues to replace something that a mail server will do much more easily is pointless. That said, it's still quite reasonable to store in a queue the fact that you need to send somebody something in the near future, and to make a queue consumer that uses those queued jobs to create and send messages. This could be a useful thing to do if your messages take a long time to generate, you want to maximise responsiveness, or allow scheduling of messages for future delivery. So instead of this:

```
Process 1: script -> create email message -> store in queue
Process 2:           get message from queue -> submit via SMTP
```
do this:
```
Process 1: script -> create queued message job
Process 2:           get queued job -> create email message -> submit via SMTP
```
Both decouple job creation from sending, but the latter doesn't require that you send a possibly out-of-date message. Sending through an SMTP relay is more like the first route, but has less overhead because it's already using native SMTP, plus it has built-in error handling, deferred deliveries, retries, bounce handling that a naïve queueing system knows nothing about:
```
Process 1: script -> create email message -> submit via local SMTP
Process 2:           mail server gets message from queue -> submit via remote SMTP
```
Also, you don't have to write process 2 here – it's what the mail server does out of the box.

As for queuing systems, I've had generally bad experiences with running rabbitmq, however, I've found [beanstalkd](https://beanstalkd.github.io/) easy to use and extremely reliable. It's not nearly as sophisticated as rabbitmq, and misses many of its features, however, it does the basics of queueing extremely well and has a great PHP client.
